### ISSUE TITLE
SessionID max length (int)

### ISSUE THREAD

**Issue by shubhank008** [bshao001/ChatLearner#68]
What's the max allowed value for sessionId or its length ? Seems if I am sending single digits (1,2,3) its working fine but sending a large id (1279856) makes it use its own sessionId instead.  

PS: This is in context to the chatserver (REST)

**Comment by bshao001** [comment]
An interesting question. However, the allowed session ID should be much bigger than the number you were mentioning. There is no explicit type definition in python for the ID, but it should be int type, and the maximum number should be much bigger than the number with which amount of sessions that any server can afford due to the system load.

The session ID cannot be explicitly assigned from the client side, it is auto increased from the server side. That may answer your question, but I am not sure what exactly you did in your experiment.

**Comment by shubhank008** [comment]
> An interesting question. However, the allowed session ID should be much bigger than the number you were mentioning. There is no explicit type definition in python for the ID, but it should be int type, and the maximum number should be much bigger than the number with which amount of sessions that any server can afford due to the system load.
> 
> The session ID cannot be explicitly assigned from the client side, it is auto increased from the server side. That may answer your question, but I am not sure what exactly you did in your experiment.

Yep, the last part answers it, I was trying to explicitly  assign a session ID to try and recognize my and my friend's userid (random numbers).  I do think that explicitly assignable session ids might be better as they would allow much larger integration of existing users/cms/etc. but I can understand its out of scope for this project and something one should learn to do himself 🥇 

PS: I do was able to make it work how I wanted by assigning the given session id to my or my friend's username. (on first chat I send 0, then I store the returned session ID to my name and in subsequent calls I send that session id instead).  

Offtopic, but I wanted to train the bot to a particular sub-reddit (to match its focused content) like that of anime or games or dota. How long should I expect to train it using GPU (1070 or K80 via cloud).  
Is there any particular epoch or learning rate indicator I should keep my eye for while training to know its ready ? 
And I read you suggested just 1 month's data, I was thinking about 6-12 months.  

Or do you think its better to train it using focused data websites like wikipedia etc. to train it for a particular topic or knowledge ?

**Comment by shubhank008** [comment]
@bshao001 Also, other than above query, if I am to create/add new rules (for jokes, stories, etc.), am I required to train whole model again, from scratch ?

**Comment by bshao001** [comment]
Concerning the training time, I cannot give you any good advice, as it depends on many factors, including model size, vocab size, sequence lengths, training data size, and the learning rates you use. If the model size is suitable to all others, you should stop at somewhere you see the perplexity starting to drop very slowly.

Yes, you have to train the model from scratch, when the training data set is updated, especially when the vocab set is modified.
